File: src/app/browser/main/tab/page/navigation/history/back.cpp
#include "back.hpp"

using namespace app::browser::main::tab::page::navigation::history;

Back::Back(
    const Glib::RefPtr<Gio::SimpleAction> & ACTION__BACK
) {
    // Init actions
    action__back = ACTION__BACK;

    // Init widget
    set_icon_name(
        "go-previous-symbolic"
    );

    set_tooltip_text(
        _("Back")
    );

    set_sensitive(
        false
    );

    signal_clicked().connect(
        [this]
        {
            action__back->activate();
        }
    );
}

void Back::update(
    const bool & ENABLED
) {
    set_sensitive(
        ENABLED
    );

    action__back->set_enabled(
        ENABLED
    );
}

File: src/browser/main/tab/page/navigation/history/back.rs
use gtk::{gio::SimpleAction, Button};
use std::sync::Arc;

pub struct Back {
    widget: Button,
}

impl Back {
    // Construct
    pub fn new(action_tab_page_navigation_history_back: Arc<SimpleAction>) -> Self {
        Self {
            widget: Button::builder()
                .icon_name("go-previous-symbolic")
                .tooltip_text("Back")
                .sensitive(false)
                .build(),
        }
    }

    // Actions
    pub fn update(&self) {
        // @TODO
    }

    // Getters
    pub fn widget(&self) -> &Button {
        &self.widget
    }
}
